sympy/functions/elementary/trigonometric.py

```diff
@@ -2251,6 +2253,12 @@ def _eval_as_leading_term(self, x):
         else:
             return self.func(arg)
 
+    def _eval_is_real(self):
+        x = self.args[0]
+        if x.is_real is False:
+            return False
```

Comment:
I expect, this is correct for complex x *and* not real.  But is_real==False doesn't imply that x.is_complex is True!